tests/unit/recommenders/models/test_sasrec_model.py

```diff
@@ -0,0 +1,225 @@
+                items = [item_dict[x[0]] for x in items]
+                for i, t in zip(items, timestamps):
+                    out_txt = [str(user_count), str(i)]
+                    if add_time:
+                        out_txt.append(str(t))
+                    fw.write(sep.join(out_txt) + "\n")
+                user_count += 1
+
+    print(f"Total {user_count-1} users, {count_del} removed")
+    print(f"Processed model input data in {pname}")
+    return user_dict, item_dict
+
+
+@pytest.mark.gpu
+def test_prepare_data():
+    data_dir = os.path.join("..", "..", "tests", "resources", "deeprec", "sasrec")
+    dataset = "reviews_Electronics_5"
+    reviews_name = dataset + ".json"
+    outfile = os.path.join(data_dir, dataset + ".txt")
+
+    reviews_file = os.path.join(data_dir, reviews_name)
+    download_and_extract(reviews_name, reviews_file)
+    reviews_output = _reviews_preprocessing(reviews_file)
+    _, _ = data_process_with_time(reviews_output, outfile, K=10, sep="\t")
+
+    # initiate a dataset class
+    data = SASRecDataSet(filename=outfile, col_sep="\t")
+
+    # create train, validation and test splits
+    data.split()
+
+    assert len(data.user_train) > 0
+    assert len(data.user_valid) > 0
+    assert len(data.user_test) > 0
+
+
+@pytest.mark.gpu
+def test_sampler():
+    batch_size = 8
+    maxlen = 50
+    data_dir = os.path.join("..", "..", "tests", "resources", "deeprec", "sasrec")
+    dataset = "reviews_Electronics_5"
+    reviews_name = dataset + ".json"
+    outfile = os.path.join(data_dir, dataset + ".txt")
+
+    reviews_file = os.path.join(data_dir, reviews_name)
+    download_and_extract(reviews_name, reviews_file)
+    reviews_output = _reviews_preprocessing(reviews_file)
+    _, _ = data_process_with_time(reviews_output, outfile, K=10, sep="\t")
+
+    # initiate a dataset class
+    data = SASRecDataSet(filename=outfile, col_sep="\t")
+
+    # create train, validation and test splits
+    data.split()
+
+    sampler = WarpSampler(
+        data.user_train,
+        data.usernum,
+        data.itemnum,
+        batch_size=batch_size,
+        maxlen=maxlen,
+        n_workers=3,
+    )
+    u, seq, pos, neg = sampler.next_batch()
+
+    assert len(u) == batch_size
+    assert len(seq) == batch_size
+    assert len(pos) == batch_size
+    assert len(neg) == batch_size
+
+
+@pytest.mark.gpu
+def test_sasrec():
+    # Amazon Electronics Data
+    itemnum = 85930
+    maxlen = 50
+    num_blocks = 2
+    hidden_units = 100
+    num_heads = 1
+    dropout_rate = 0.1
+    l2_emb = 0.0
+    num_neg_test = 100
+
+    model = SASREC(
+        item_num=itemnum,
+        seq_max_len=maxlen,
+        num_blocks=num_blocks,
+        embedding_dim=hidden_units,
+        attention_dim=hidden_units,
+        attention_num_heads=num_heads,
+        dropout_rate=dropout_rate,
+        conv_dims=[100, 100],
+        l2_reg=l2_emb,
+        num_neg_test=num_neg_test,
+    )

```

Comment:
An elegant way of adding input data to tests are fixtures, see how they work here: https://github.com/miguelgfierro/pybase/blob/master/test/pytest_fixtures.py
you could create a dictionary of fixtures of Amazon electronics data and input it in the test